### ISSUE TITLE
makeDesktopItem files aren't executable

### ISSUE THREAD

**Issue by LunNova** [NixOS/nixpkgs#152059]
KDE won't run .desktop files that aren't executable, but this usually doesn't matter because it also allows running files owned by root and these are.

https://github.com/KDE/kdelibs/blob/1c5b25966136b6099665477616fb4d38a98fd7ef/kdecore/config/kdesktopfile.cpp#L170-L174

Unfortunately if you are using a buildFHSUserEnv which uses bubblewrap, it will show files owned by root as being owned by nobody.

So if you launch a game inside lutris or steam-run, and then that game runs xdg-open to open a web browser, and the .desktop entry for that browser is not executable but is owned by root KDE will no longer be able to tell it's owned by root and will pop up this error.

![image](https://user-images.githubusercontent.com/782440/147321139-4d4cfbc9-2fad-4e7c-847a-c3142846302b.png)

Clicking yes can't work as it's in the nix store.

![image](https://user-images.githubusercontent.com/782440/147321290-4657d5ed-f8d1-42a6-a38e-ade5ef677ce4.png)

### Steps To Reproduce
Use pkgs.makeDesktopItem to make a .desktop file, check its executable bit.

```
$ ls -l /nix/store/dk0rv65i41kzsh991s2cj0v8wkkdg5vs-firefox.desktop/share/applications/firefox.desktop
-r--r--r-- 2 root root 203 Dec 31  1969 /nix/store/dk0rv65i41kzsh991s2cj0v8wkkdg5vs-firefox.desktop/share/applications/firefox.desktop
```

### Expected behavior
$ ls -l /nix/store/dk0rv65i41kzsh991s2cj0v8wkkdg5vs-firefox.desktop/share/applications/firefox.desktop
-r-xr-xr-x 2 root root 203 Dec 31  1969 /nix/store/dk0rv65i41kzsh991s2cj0v8wkkdg5vs-firefox.desktop/share/applications/firefox.desktop

A `#!/usr/bin/env xdg-open` shebang [should probably also be added.](https://unix.stackexchange.com/questions/373239/what-is-the-advantage-of-desktop-files-without-executable-bit-set)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
lun@hisame ~ ./rw
$ nix-info -m
 - system: `0`
 - host os: `Linux 5.15.10, NixOS, 22.05 (Quokka)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.0pre20211217_6e6e998`
 - channels(root): `"nixos-22.05pre336298.bc5d68306b4"`
 - nixpkgs: `0`
```

**Comment by LunNova** [comment]
Not sure who best to ping, pinging a few people who have touched pkgs/build-support/make-desktopitem/default.nix, hope this is okay.

Happy to PR a fix if this sounds okay to implement. Not sure this actually counts as a problem since the real problem might be more that xdg-open shouldn't need to work correctly running under some random app's FHS sandbox.

@necessarily-equal @flokli @LogicalOverflow

**Comment by necessarily-equal** [comment]
Making desktop files executable sound like a good idea indeed, the 
special case where deskop files owned by root are alowed to be run 
anyway is a hack, IMO. I'm not a desktop file master, but I'd welcome a 
fix that makes them executable (add the +x mod).

**Comment by vcunat** [comment]
Fixing `makeDesktopItem` won't suffice, I'm afraid, though I don't have an estimate how large fraction would remain.  Maybe a stdenv-default hook similar to this one?
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/prune-libtool-files.sh

**Comment by jtojnar** [comment]
I think marking files produced with `makeDesktopItem` executable is fine but I would be wary of making random files executable. The Nautilus CVE mentioned in https://unix.stackexchange.com/questions/373239/what-is-the-advantage-of-desktop-files-without-executable-bit-set appears to be fixed (the desktop file is listed using file name, not the `Name` key from the file) but who knows what else can go wrong.

**Comment by LunNova** [comment]
Closed in favor of #160923.
